server/src/main/java/io/crate/protocols/postgres/Messages.java

```diff
@@ -179,40 +173,34 @@ static void sendParameterStatus(Channel channel, final String name, final String
     static void sendAuthenticationError(Channel channel, String message) {
         LOGGER.warn(message);
         byte[] msg = message.getBytes(StandardCharsets.UTF_8);
-        sendErrorResponse(channel, message, msg, SEVERITY_FATAL, null, null,
-            METHOD_NAME_CLIENT_AUTH, ERROR_CODE_INVALID_AUTHORIZATION_SPECIFICATION);
+        byte[] errorCode = PGErrorStatus.INVALID_AUTHORIZATION_SPECIFICATION.code().getBytes(StandardCharsets.UTF_8);
+
+        sendErrorResponse(channel, message, msg, PGError.SEVERITY_FATAL, null, null,
+                          METHOD_NAME_CLIENT_AUTH, errorCode);
     }
 
     static ChannelFuture sendErrorResponse(Channel channel, Throwable throwable) {
-        final String message = SQLExceptions.messageOf(throwable);
-        byte[] msg = message.getBytes(StandardCharsets.UTF_8);
+        var error = PGError.fromThrowable(throwable);
+        byte[] msg = error.message().getBytes(StandardCharsets.UTF_8);
+        byte[] errorCode = error.status().code().getBytes(StandardCharsets.UTF_8);
         byte[] lineNumber = null;
         byte[] fileName = null;
         byte[] methodName = null;
 
-        StackTraceElement[] stackTrace = throwable.getStackTrace();
-        if (stackTrace != null && stackTrace.length > 0) {
-            StackTraceElement stackTraceElement = stackTrace[0];
-            lineNumber = String.valueOf(stackTraceElement.getLineNumber()).getBytes(StandardCharsets.UTF_8);
-            if (stackTraceElement.getFileName() != null) {
-                fileName = stackTraceElement.getFileName().getBytes(StandardCharsets.UTF_8);
-            }
-            if (stackTraceElement.getMethodName() != null) {
-                methodName = stackTraceElement.getMethodName().getBytes(StandardCharsets.UTF_8);
+        if (throwable != null) {
```

Comment:
Either the argument should be annotated with `Nullable` or this check is not needed. (In which case the `PGError.fromThrowable(..)` call earlier may be strange as well?